scenario/runtime.py

```diff
@@ -298,38 +295,48 @@ def _virtual_charm_root(self) -> typing.ContextManager[Path]:
         config_yaml = virtual_charm_root / "config.yaml"
         actions_yaml = virtual_charm_root / "actions.yaml"
 
-        metadata_files_present = any(
-            file.exists() for file in (metadata_yaml, config_yaml, actions_yaml)
-        )
+        metadata_files_present: Dict[Path, Union[str, False]] = {
+            file: file.read_text() if file.exists() else False
```

Comment:
I think `Optional[str]`, that is, `Union[str, None]`, would be more typical than `Union[str, False]`.

Reply:
mmm yeah, I'm always a bit hesitant to give None a falsy semantics because of how slippery it is on python, but I guess it's more legible